rust/core-lib/src/view.rs

```diff
@@ -106,6 +110,22 @@ struct DragState {
     max: usize,
 }
 
+/// Information about search queries and number of matches for find
+#[derive(Serialize, Deserialize, Debug)]
+pub struct FindStatus {
+    /// The current search query.
+    chars: Option<String>,
+
+    /// Whether the active search is case matching.
+    case_sensitive: Option<bool>,
+
+    /// Whether the search query is considered as regular expression.
+    is_regex: Option<bool>,
+
+    /// Total number of matches.
+    matches: usize
```

Comment:
do we also want this to include the index of the currently active highlight?

Reply:
If there are multiple queries then only one active highlight would exist for one of the `Find`. It might make sense to get the index if all occurrences of all queries are considered, however, that will be a bit tricky. It also seems that most editors only show the total number of matches.